src/main/java/com/atlassian/bitbucket/jenkins/internal/model/BitbucketBuildStatus.java

```diff
@@ -0,0 +1,50 @@
+package com.atlassian.bitbucket.jenkins.internal.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import hudson.model.AbstractBuild;
+import okhttp3.HttpUrl;
+import org.jenkinsci.plugins.displayurlapi.DisplayURLProvider;
+
+import java.net.URL;
+
+public class BitbucketBuildStatus {
+
+    private final String description;
+    private final String key;
+    private final String name;
+    private final BuildState state;
+    private final URL url;
+
+    public BitbucketBuildStatus(AbstractBuild build) {
+        state = BuildState.fromBuild(build);
+        description = state.getDescriptiveText(build);
+        key = build.getId();
+        name = build.getProject().getName();
+        url = HttpUrl.parse(DisplayURLProvider.get().getRunURL(build)).url();
```

Comment:
This solution is imperfect, as it appends `/display/redirect` to the URL, which is valid but unnecessary.

Reply:
Also, I kept it as a URL here, but seeing as it's being unpacked into a String anyway, this could just as easily be a String to start with.

Reply:
how about `Jenkins.get().getRootUrl() + "/" +build.getUrl()` ? does that not work?

Reply:
I like types in general, so I generally prefer to not have stringly typed things. If we can keep it without too much cost, I say we should.